R/robustbayesianmetaanalysis.R

```diff
@@ -0,0 +1,2419 @@
+            options$input_N1, options$input_N2
+          ) != "")
+        )
+    }
+  } else if (options$measures == "correlation") {
+    ready_arg2 <-
+      any(c(options$input_CI, options$input_SE, options$input_N) != "",
+          CI_input)
+  } else if (options$measures == "general") {
+    ready_arg2 <-
+      any(c(options$input_CI, options$input_SE) != "", CI_input)
+  }
+  
+  return(ready_arg1 && ready_arg2)
+  
+}
+.RoBMA_model_notifier       <- function(jaspResults) {
+  # We don't wanna delete the RoBMA modele every time settings is change since RoBMA takes a lot of time to fit.
+  # Therefore, we don't create dependencies on the fitted model, but on a notifier that tells us when there was
+  # a change. If possible, we don't refit the whole model, just update the neccessary parts.
+  
+  if (is.null(jaspResults[["model_notifier"]])) {
+    model_notifier <- createJaspState()
+    model_notifier$dependOn(.RoBMA_dependencies)
+    jaspResults[["model_notifier"]] <- model_notifier
+  }
+  
+  return()
+  
+}
+.RoBMA_data_get             <- function(options, dataset) {
+  if (options$measures == "fitted") {
+    return(NULL)
+    
+  } else{
+    if (!is.null(dataset)) {
+      return(dataset)
+    } else{
+      var_names <-
+        c(
+          options$input_t,
+          options$input_ES,
+          options$input_SE,
+          options$input_N,
+          options$input_N1,
+          options$input_N2,
+          unlist(options$input_CI)
+        )
+      var_names <- var_names[var_names != ""]
+      
+      dataset <-
+        readDataSetToEnd(columns.as.numeric = var_names,
+                         columns = if (options$input_labels != "")
+                           options$input_labels)
+      
+      # compute SE from CI
+      if (length(options$input_CI) != 0) {
+        dataset$SE <-
+          abs(dataset[, .v(options$input_CI[[1]][1])] - dataset[, .v(options$input_CI[[1]][2])]) / (2 *
+                                                                                                      1.96)
+      }
+    }
+    
+  }
+  
+  return(dataset)
+}
+.RoBMA_priors_get           <- function(jaspResults, options) {
+  if (!is.null(jaspResults[["priors"]])) {
+    return()
+  } else{
+    priors <- createJaspState()
+    priors$dependOn(.RoBMA_dependencies)
+    jaspResults[["priors"]] <- priors
+  }
+  
+  priors_mu    <- NULL
+  priors_tau   <- NULL
+  priors_omega <- NULL
+  
+  priors_mu_null    <- NULL
+  priors_tau_null   <- NULL
+  priors_omega_null <- NULL
+  
+  
+  # add priors from options
+  if (length(options$priors_mu) != 0) {
+    for (i in 1:length(options$priors_mu)) {
+      priors_mu <- c(priors_mu,
+                     list(.RoBMA_options2priors(options$priors_mu[[i]])))
+    }
+  }
+  
+  if (length(options$priors_tau) != 0) {
+    for (i in 1:length(options$priors_tau)) {
+      priors_tau <- c(priors_tau,
+                      list(.RoBMA_options2priors(options$priors_tau[[i]])))
+    }
+  }
```

Comment:
something like this
```r
object <- list()
prior_elements <- c("priors_mu", "priors_omega", ...)
prior_names <- c("mu", "omega", ...)
for (i in seq_along(prior_elements)) {
  tmp <- NULL
  for (elem in options[[prior_elements[i]]]) {
     tmp <- c(tmp, list(.RoBMA_options2priors(elem)))
  }
  object[[prior_names[i]]] <- tmp
}
```
is equivalent to the 6 for loops right? if you use `seq_along` or `elem in ...` then the empty lists are automatically skipped.

Reply:
That's very clever! Thanks, I changed it accordingly.